llmspec/llmspec.py

```diff
@@ -221,19 +215,24 @@ class ChatResponse(LMResponse):
     choices: List[ChatChoice]
 
 
-class EmbeddingRequest(msgspec.Struct):
+class EmbeddingRequest(msgspec.Struct, JSONSerializableMixin):
     model: str
     input: Union[str, List[str]]
     user: str = ""
 
 
 class EmbeddingData(msgspec.Struct):
-    embedding: List[float]
+    # not using EmbeddingValue
+    # TypeError: Type unions may not contain more than
+    # one array-like (list, set, tuple) type - type
+    # `typing.Union[typing.List[float], typing.List[typing.List[float]]]`
+    # is not supported
+    embedding: List[Union[float, List[float]]]
```

Comment:
This is not the correct type we expect. Since it can be `[1.0, [2.0]]`.

I will create another issue to track this.